plugins/jobs/src/js/data/JobModel.ts

```diff
@@ -524,17 +527,98 @@ const fieldResolvers = {
   }
 };
 
+// Sort and filtering
+function sortJobs(
+  jobs: Job[],
+  sortBy: SortOption = "ID",
+  sortDirection: SortDirection = "ASC"
+): Job[] {
+  jobs.sort((a, b) => {
+    let result;
+
+    switch (sortBy) {
+      case "ID":
+        result = sortJobById(a, b);
+        break;
+      case "STATUS":
+        result = sortJobByStatus(a, b);
+        break;
+      case "LAST_RUN":
+        result = sortJobByLastRun(a, b);
+        break;
+      default:
+        result = 0;
+    }
+
+    const direction = sortDirection === "ASC" ? 1 : -1;
+    return result * direction;
+  });
+
+  return jobs;
+}
+
+function sortJobById(a: Job, b: Job): number {
+  return a.id.localeCompare(b.id);
+}
+
+function sortJobByStatus(a: Job, b: Job): number {
+  return (
+    JobStates[a.scheduleStatus].sortOrder -
+    JobStates[b.scheduleStatus].sortOrder
+  );
+}
+
+function sortJobByLastRun(a: Job, b: Job): number {
+  return (
+    JobStatus[a.lastRunStatus.status].sortOrder -
+    JobStatus[b.lastRunStatus.status].sortOrder
+  );
+}
+
+function filterJobs(
+  jobs: MetronomeClient.JobResponse[],
+  filter: string | null
+): MetronomeClient.JobResponse[] {
+  if (filter === null) {
+    return jobs;
+  }
+
+  return jobs.filter(({ id }) => id.indexOf(filter) !== -1);
+}
+
 export const resolvers = ({
+  fetchJobs,
   fetchJobDetail,
   pollingInterval
 }: ResolverArgs): IResolvers => ({
   Query: {
     jobs(
-      _obj = {},
-      _args: GeneralArgs = {},
+      _parent = {},
+      args: GeneralArgs = {},
       _context = {}
-    ): Observable<JobConnection | null> {
-      return Observable.of(null);
+    ): Observable<Job[] | null> {
+      const {
+        sortBy = "ID",
+        sortDirection = "ASC",
+        filter = null
+      } = args as JobsQueryArgs;
+
+      const pollingInterval$ = Observable.timer(0, pollingInterval);
+      const responses$ = pollingInterval$.switchMap(fetchJobs);
+
+      return (
+        responses$
+          .map(jobs => filterJobs(jobs, filter))
+          // N+1 request madness
+          .map(jobs => jobs.map(job => fetchJobDetail(job.id)))
+          .switchMap(obs => Observable.combineLatest(obs))
```

Comment:
what is this doing? 🤔

Reply:
There is a fiddle here where you can play with the code:
https://jsfiddle.net/fabsn/x5zaf129/23/

The main idea is, you have the first Observable<Jobs[]> (an observable that emits jobs). So you have to map in a request for every job. That is what line 613 makes `Jobs[] => Observable<JobsWithDetail>[]`. But a list of Observables (and not an observable of lists) is not very interesting because the list data structure is boring :). So we use Observable.combineLatest to convert `Observable<JobsWithDetail>[] => Observable<JobsWithDetail[]>`, this is line 614 map part. 

At this step you Have an Observable<Observable<JobsWithDetail[]>>, and you need switchMap (or other strategy) to remove the nesting. This is the switch part of line 614.